pyteal/ast/abi/address_test.py

```diff
@@ -0,0 +1,76 @@
+from ... import *
+
+options = CompileOptions(version=5)
+
+
+def test_AddressTypeSpec_str():
+    assert str(abi.AddressTypeSpec()) == "address"
+
+
+def test_AddressTypeSpec_is_dynamic():
+    assert (abi.AddressTypeSpec()).is_dynamic() is False
+
+
+def test_AddressTypeSpec_byte_length_static():
+    assert (abi.AddressTypeSpec()).byte_length_static() == abi.ADDRESS_LENGTH
+
+
+def test_AddressTypeSpec_new_instance():
+    assert isinstance(abi.AddressTypeSpec().new_instance(), abi.Address)
+
+
+def test_AddressTypeSpec_eq():
+    assert abi.AddressTypeSpec() == abi.AddressTypeSpec()
+
+    for otherType in (
+        abi.ByteTypeSpec,
+        abi.StaticArrayTypeSpec(abi.ByteTypeSpec(), 31),
+        abi.DynamicArrayTypeSpec(abi.ByteTypeSpec()),
+    ):
+        assert abi.AddressTypeSpec() != otherType
+
+
+def test_Address_encode():
+    value = abi.Address()
+    expr = value.encode()
+    assert expr.type_of() == TealType.bytes
+    assert expr.has_return() is False
```

Comment:
That's how it was before... I asked for this version because `has_return()` is boolean so you can pin the value precisely without worrying about truthiness.

Reply:
(it's not really a big deal. In theory, we've tested that `has_return()` is boolean elsewhere and works as expected so that using the more concise _truthy_ version is just as good)

Reply:
I see, you're concerned that `not expr.has_return()` would also appear true if `expr.has_return()` returned some other false-y value, like 0? That's something I've not considered before...

I suppose you'd also be in favor of `assert X is True` over `assert X` if we wanted to test that a value is _really_ true, not merely truth-y?

I don't love the syntax but the argument makes a lot of sense

Reply:
so going with is False?

Reply:
Just to answer @jasonpaulos - yeah. I would also use `is True` or `X is None` instead of `not X` ... etc ...

Reply:
Thanks for the discussion, let's keep it as `is False` then.